src/renderer/curate/importCuration.ts

```diff
@@ -151,19 +151,28 @@ const validMetaNames = ['meta.txt', 'meta.yaml', 'meta.yml'];
  * @param dir Path to search
  */
 async function getRootPath(dir: string): Promise<string | undefined> {
-  const files = await fs.readdir(dir);
-  for (let i = 0; i < files.length; i++) {
-    const file = files[i];
-    const fullPath = path.join(dir, file);
-    const stats = await fs.lstat(fullPath);
-    // Found root, pass back
-    if (stats.isFile() && endsWithList(file.toLowerCase(), validMetaNames)) {
-      return dir;
-    } else if (stats.isDirectory()) {
-      const foundRoot = await getRootPath(fullPath);
-      if (foundRoot) {
-        return foundRoot;
+  // Start off by initializing the queue to the contents of the root directory.
+  const queue = await fs.readdir(dir);
+  // As long as there are still queue items to read, we keep going.
+  // Note: if we find the item, we will return before this completes.
+  while (queue.length != 0) {
+    // Pop the first element. We'll need to combine it with the root path to get
+    // the full path.
+    const entry = queue.shift();
+    // @ts-ignore
+    const fullpath = path.join(dir, entry);
```

Comment:
The bot is complaining about the `// @ts-ignore`. TS gets worried that `entry` might be undefined. `entry` will not be undefined, because we checked that the `queue` length wasn't 0 *right* before taking `entry` from the front of `queue`. TS is wrong and shouldn't worry.